File: frontend/src/api/db-table-group.ts
import { request } from '@/utils/request'

// 分组接口
export interface DbTableGroup {
  id?: number
  parentId?: number
  groupName: string
  groupCode?: string
  description?: string
  sortOrder?: number
  createTime?: string
  updateTime?: string
  children?: DbTableGroup[]
  tableCount?: number
}

// 获取分组树
export const getGroupTree = () => {
  return request.get<DbTableGroup[]>('/db-table-group/tree')
}

// 创建分组
export const createGroup = (data: DbTableGroup) => {
  return request.post<number>('/db-table-group', data)
}

// 更新分组
export const updateGroup = (id: number, data: DbTableGroup) => {
  return request.put(`/db-table-group/${id}`, data)
}

// 删除分组
export const deleteGroup = (id: number) => {
  return request.delete(`/db-table-group/${id}`)
}

File: backend/src/main/java/com/lowcode/entity/DbTableGroup.java
package com.lowcode.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 库表分组实体类
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("db_table_group")
public class DbTableGroup implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 父分组ID，0表示顶级分组
     */
    private Long parentId;

    /**
     * 分组名称
     */
    private String groupName;

    /**
     * 分组编码（可选）
     */
    private String groupCode;

    /**
     * 分组描述
     */
    private String description;

    /**
     * 排序号
     */
    private Integer sortOrder;

    /**
     * 是否删除
     */
    @TableLogic
    private Integer deleted;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 子分组列表（非数据库字段）
     */
    @TableField(exist = false)
    private List<DbTableGroup> children;

    /**
     * 分组下的表数量（非数据库字段）
     */
    @TableField(exist = false)
    private Integer tableCount;
}
